docs/src/references/agent/device-management-api.md

```diff
@@ -0,0 +1,330 @@
+  - An operation might define specific `"status"` values for extra steps, actions or checks.
+
+:::info
+Health checks is a notable exception to the rule "command status are published as retained messages".
+A health check is sent to the channel `cmd/health/check` of a service (or a device) as a non-retained request `{}`,
+triggering a health message response published on the `status/health` channel of the same service
+(or respectively the channels of all the service running on that device).
+:::
+
+## Operation workflow
+
+A specific workflow rules each operation type, with specific:
+- states
+- message payloads
+- states transitions.
+
+However, there are some common rules:
+- There at least four states: `"init"`, `"executing"`, `"successful"` and `"failed"`.
+- A new command has to be created in the `"init"` state.
+- Some checks can be done before the `"executing"` state, but this one should be the first triggering updates on the system.
+- There only two terminal states: `"successful"` and `"failed"`.
+- The retained messages are finally cleared by the requester.
+
+```mermaid
+flowchart LR
+    init --> executing --> successful
+    executing --> failed
+    successful --> done
+    failed --> done
+
+    subgraph init_state["status: init"]
+      init(gives all the data to proceed<br/><br/>created by the requester<br/>typically a mapper)
+    end
+
+    subgraph executing_state["status: executing"]
+      executing(the command is under execution<br/>the content is the same as for init)
+    end
+
+    subgraph successful_state["status: successful"]
+      successful(the command has been successful<br/>some outcome might be returned to the requester)
+    end
+
+    subgraph failed_state["status: failed"]
+      failed(the command has failed<br/>the reason is given to the requester)
+    end
+
+    subgraph done_state[done]
+      done(finally cleared by the requester)
+    end
+```
+
+
+### Capability declaration
+
+On start, a service that implements an operation for a device must:
+
+- know the MQTT root prefix (per default: `te`),
+- know the entity topic identifier of the target device (e.g. `device/child001//`),
+- know the well-known name of the operation (e.g. `software-update`),
+- forge a capability message describing the operation support (e.g. `{ "supportedTypes": ["apt", "docker"] }`)
+- publish this capability message, with a retained flag, on the target command metadata topic (e.g. `te/device/child001///cmd/software-update`)
+
+### Command execution
+
+To request the execution of a command, a client must:
+- know the MQTT root prefix (per default: `te`),
+- know the entity topic identifier of the target device (e.g. `device/child001//`),
+- know the well-known name of the operation (e.g. `software-update`),
+- assign a unique id to its command (say `c8y-1234`)
+- use these target and command identifiers to forge the topic name for its request
+  (e.g. `te/device/child001///cmd/software-update/c8y-1234`)
```

Comment:
I'd just compress all that into a single statement: "subscribe to the command topic that includes the command id as well (e.g. `te/device/child001///cmd/software-update/c8y-1234`).

Using a term like command topic should be sufficient as we've already established its format in the previous section.

Reply:
Agree that this is bit verbose. But I think it helps to be explicit in a reference documentation on what needs to given to and then done by the operation plugins.